shell/platform/windows/flutter_windows_engine_unittests.cc

```diff
@@ -409,5 +419,51 @@ TEST(FlutterWindowsEngine, GetExecutableName) {
   EXPECT_EQ(engine->GetExecutableName(), "flutter_windows_unittests.exe");
 }
 
+// Ensure that after setting or resetting the high contrast feature,
+// the corresponding status flag can be retrieved from the engine.
+TEST(FlutterWindowsEngine, UpdateHighContrastFeature) {
+  std::unique_ptr<FlutterWindowsEngine> engine = GetTestEngine();
+  EngineModifier modifier(engine.get());
+
+  modifier.embedder_api().Run = MOCK_ENGINE_PROC(
+      Run, ([](auto version, auto config, auto args, auto user_data,
+               FLUTTER_API_SYMBOL(FlutterEngine) * engine_out) {
+        *engine_out = reinterpret_cast<FLUTTER_API_SYMBOL(FlutterEngine)>(1);
+        return kSuccess;
+      }));
+  modifier.embedder_api().NotifyDisplayUpdate = MOCK_ENGINE_PROC(
+      NotifyDisplayUpdate,
+      ([](auto engine, auto type, auto data, auto size) { return kSuccess; }));
+  modifier.embedder_api().UpdateLocales = MOCK_ENGINE_PROC(
+      UpdateLocales, ([](auto engine, const FlutterLocale** locales,
+                         size_t locales_count) { return kSuccess; }));
+  modifier.embedder_api().SendPlatformMessage =
+      MOCK_ENGINE_PROC(SendPlatformMessage,
+                       ([](auto engine, auto message) { return kSuccess; }));
```

Comment:
Is all this necessary? Can we remove this, plus the `engine->Run()` and `Shutdown` below?

Reply:
Oh I see. Could we drop the `if (engine_)` check to make testing easier? For example, here's some prior art:

https://github.com/flutter/engine/blob/77648b90b73baf6d77c02bf697b64acd9a6bcc18/shell/platform/windows/flutter_windows_engine.cc#L387-L394

Reply:
Actually yes, I think that should be fine since the non-mocked embedder API's `UpdateAccessibilityFeatures` performs its own check for when the engine is null.